### ISSUE TITLE
PyTorch Model Config Autocomplete

### ISSUE THREAD

**Issue by trevoryao** [triton-inference-server/server#8233]
**Is your feature request related to a problem? Please describe.**
We are looking to speed-up Pytorch model onboarding and also have an ultimate independent source of truth for the "schema" of inputs into the Triton Server. We have run into issues loading torch scripted models with how the type inference system infers types of Dictionary of tensors input format, during input/output format verification.

**Describe the solution you'd like**
We'd like to be able to take advantage of input/output autocomplete of the Triton model config, like we can with the Tensorflow/ONNX backends.

**Comment by dzier** [comment]
Unfortunately, the TorchScript format for the libtorch in pytorch backend doesn't carry sufficient metadata to learn what are inputs/outputs for the model. This means that Triton cannot support auto-complete configs for pytorch backend. There was some discussion at one point to support TorchScript Model Archive format, which does carry this information, but it was never prioritized.

If you would like to contribute solution, we would be happy to review and provide guidance.
